plumbing/http/http.go

```diff
@@ -0,0 +1,327 @@
+package http
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-billy/v5/osfs"
+	"github.com/go-git/go-git/v6/plumbing/storer"
+	"github.com/go-git/go-git/v6/plumbing/transport"
+	"github.com/go-git/go-git/v6/storage"
+	"github.com/go-git/go-git/v6/utils/ioutil"
+)
+
+type contextKey string
+
+type service struct {
+	method  string
+	handler http.HandlerFunc
+	svc     transport.Service
+}
+
+var services = map[string]service{
```

Comment:
Maybe move services as a default into config? We should be able to add/remove services.

Reply:
This is meant to implement the default services and be sort of a replacement for `git-http-backend`. It's just a middleware, other services can be defined as handlers and wrap this one.

Reply:
Ok, what about disabling dumb protocol routes?